framework/src/modules/dpos/transaction_assets/pom_transaction_asset.ts

```diff
@@ -109,34 +110,8 @@ export class PomTransactionAsset extends BaseAsset<PomTransactionAssetContext> {
 			throw new Error('BlockHeaders are identical. No contradiction detected.');
 		}
 
-		/*
-			Check for BFT violations:
-					1. Double forging
-					2. Disjointedness
-					3. Branch is not the one with largest maxHeightPrevoted
-		*/
-
-		let b1 = asset.header1;
-		let b2 = asset.header2;
-
-		// Order the two block headers such that b1 must be forged first
-		if (
-			b1.asset.maxHeightPreviouslyForged > b2.asset.maxHeightPreviouslyForged ||
-			(b1.asset.maxHeightPreviouslyForged === b2.asset.maxHeightPreviouslyForged &&
-				b1.asset.maxHeightPrevoted > b2.asset.maxHeightPrevoted) ||
-			(b1.asset.maxHeightPreviouslyForged === b2.asset.maxHeightPreviouslyForged &&
-				b1.asset.maxHeightPrevoted === b2.asset.maxHeightPrevoted &&
-				b1.height > b2.height)
-		) {
-			b1 = asset.header2;
-			b2 = asset.header1;
-		}
-
-		if (
-			!(b1.asset.maxHeightPrevoted === b2.asset.maxHeightPrevoted && b1.height >= b2.height) &&
-			!(b1.height > b2.asset.maxHeightPreviouslyForged) &&
-			!(b1.asset.maxHeightPrevoted > b2.asset.maxHeightPrevoted)
-		) {
+		// Check for BFT violations:
+		if (!areHeadersContradicting(asset.header1, asset.header2)) {
```

Comment:
do we first need to check if the blockheaders are ordered correctly?

Reply:
ordering is taken care in the function https://github.com/LiskHQ/lisk-sdk/pull/5897/files#diff-9b023404122bb142e9cfbdaeb19057ed4d81383f2f6d253d0a78f2b78722fa9cR18-R34